pkgs/by-name/pi/pihole/package.nix

```
@@ -0,0 +1,257 @@
+        "$utilsfile" = [ "${scriptsDir}/utils.sh" ];
+        "$apifile" = [ "${scriptsDir}/api.sh" ];
+        "$piholeGitDir" = [ "${out}/usr/share/pihole" ];
+        "$PIHOLE_COMMAND" = [ "pihole" ];
+      };
+      keep = {
+        source = [
+          "$pihole_FTL" # Global config file
+          "$setupVars" # Global config file
+          "$PIHOLE_SETUP_VARS_FILE"
+          "$versionsfile" # configuration file, doesn't exist on NixOS
+          "${out}/usr/share/pihole/automated install/basic-install.sh"
+          "${scriptsDir}/COL_TABLE"
+          "${scriptsDir}/database_migration/gravity-db.sh"
+          "${scriptsDir}/gravity.sh"
+          "${scriptsDir}/piholeCheckout.sh"
+          "${scriptsDir}/utils.sh"
+          "${scriptsDir}/api.sh"
+          "/etc/os-release"
+          "/etc/pihole/versions"
+          "/etc/pihole/setupVars.conf"
+          "$cachedVersions"
+        ];
+
+        "$PIHOLE_SETUP_VARS_FILE" = true;
+        "$PKG_INSTALL" = true; # System package manager, patched out
+        "$PKG_MANAGER" = true; # System package manager, patched out
+        "$cmd" = true; # ping or ping6
+        "$program_name" = true; # alias for $1
+        "$svc" = true; # dynamic restart command
+        "${out}/bin/pihole" = true;
+        "${scriptsDir}/api.sh" = true;
+        "${scriptsDir}/gravity.sh" = true;
+        "${scriptsDir}/list.sh" = true;
+        "${scriptsDir}/piholeARPTable.sh" = true;
+        "${scriptsDir}/piholeDebug.sh" = true;
+        "${scriptsDir}/piholeLogFlush.sh" = true;
+        "${scriptsDir}/query.sh" = true;
+        "${scriptsDir}/uninstall.sh" = true;
+        "${scriptsDir}/update.sh" = true;
+        "${scriptsDir}/updatecheck.sh" = true;
+        "${scriptsDir}/version.sh" = true;
+
+        # boolean variables
+        "$addmode" = true;
+        "$noReloadRequested" = true;
+        "$oldAvail" = true;
+        "$verbose" = true;
+        "$web" = true;
+        "$wildcard" = true;
+
+        # Note that this path needs to be quoted due to the whitespace.
+        # TODO: raise upstream resholve issue. pihole scripts specify this path
+        # both quoted and escaped. Resholve apparently requires matching the
+        # literal path, so we need to provide a version with and without the
+        # backslash.
+        "'${out}/usr/share/pihole/automated\\ install/basic-install.sh'" = true;
+        "'${out}/usr/share/pihole/automated install/basic-install.sh'" = true;
+
+        "/etc/.pihole" = true; # Patched with an override
+        "/etc/os-release" = true;
+        "/etc/pihole/versions" = true;
+        "/etc/pihole/setupVars.conf" = true;
+      };
+      execer = [
+        "cannot:${pihole-ftl}/bin/pihole-FTL"
+        "cannot:${iproute2}/bin/ip"
+        "cannot:${systemd}/bin/systemctl"
+        "cannot:${glibc.bin}/bin/ldd"
+        "cannot:${out}/bin/pihole"
+      ];
+    };
+
+  meta = {
+    description = "A black hole for Internet advertisements";
```

```suggestion
    description = "Black hole for Internet advertisements";
```
https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#meta-attributes